client.go

```diff
@@ -293,6 +296,12 @@ func (c *client) mainloop(ctx context.Context, params *lookupParams) {
 				// This is also a point to possibly stop probing actively for a
 				// service entry.
 				params.Entries <- e
+				// DoS protection: don't cache more than maxSentEntries entries
+				if len(sentEntries) >= maxSentEntries {
+					for key := range sentEntries {
+						delete(sentEntries, key)
```

Comment:
Also: we can just store a set, right? No need to store the values.

Reply:
Otherwise, a small number of entries could still be large.

Reply:
Let's do that in a separate PR.